src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up memory cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class MemoryCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds memory caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        public static void AddMemoryCache(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Transient<IDistributedCache, MemoryDistributedCache>());
```

Comment:
As far as which options to go with: I think these are all completely separate `services.AddXyzCache()` methods. I don't think the builder pattern makes sense here because there's no concrete relationship between these services (we just happened to implement one using the other, but that's an implementation choice, not a contract obligation).